app/src/main/java/com/schneewittchen/rosandroid/widgets/location/LocationView.java

```diff
@@ -0,0 +1,225 @@
+
+    Paint buttonPaint;
+    TextPaint textPaint;
+    StaticLayout staticLayout;
+
+    protected LocationManager locationManager;
+    double gpsLatitude;
+    double gpsLongitude;
+    double gpsAltitude;
+    double networkLatitude;
+    double networkLongitude;
+    double networkAltitude;
+    long gpsTime = 0;
+    long networkTime = 0;
+
+    LocationListener locationListenerGps = new LocationListener() {
+        public void onLocationChanged(Location location) {
+            gpsTime = location.getTime();
+            gpsLatitude = location.getLatitude();
+            gpsLongitude = location.getLongitude();
+            gpsAltitude = location.getAltitude();
+            Log.d("Gps2RosView","GPS - Longitude: "+ gpsLongitude +" Latitude: " + gpsLatitude +" Altitude: " + gpsAltitude);
+            publishCoordinates();
+        }
+    };
+
+    LocationListener locationListenerNetwork = new LocationListener() {
+        public void onLocationChanged(Location location) {
+            networkTime = location.getTime();
+            networkLatitude = location.getLatitude();
+            networkLongitude = location.getLongitude();
+            networkAltitude = location.getAltitude();
+            Log.d("Gps2RosView","NETWORK - Longitude: "+networkLongitude+" Latitude: " + networkLatitude +" Altitude: " + networkAltitude);
+            publishCoordinates();
+        }
+    };
+
+    public LocationView(Context context) {
+        super(context);
+        this.context = context;
+        init();
+    }
+
+    public LocationView(Context context, @Nullable AttributeSet attrs) {
+        super(context, attrs);
+        this.context = context;
+        init();
+    }
+
+    private void init() {
+
+        buttonPaint = new Paint();
+        buttonPaint.setColor(getResources().getColor(R.color.colorPrimary));
+        buttonPaint.setStyle(Paint.Style.FILL_AND_STROKE);
+
+        textPaint = new TextPaint();
+        textPaint.setColor(Color.BLACK);
+        textPaint.setStyle(Paint.Style.FILL_AND_STROKE);
+        textPaint.setTextSize(26 * getResources().getDisplayMetrics().density);
+
+        locationManager = (LocationManager) context.getSystemService(Context.LOCATION_SERVICE);
+        if (ActivityCompat.checkSelfPermission(context, Manifest.permission.ACCESS_FINE_LOCATION) != PackageManager.PERMISSION_GRANTED && ActivityCompat.checkSelfPermission(context, Manifest.permission.ACCESS_COARSE_LOCATION) != PackageManager.PERMISSION_GRANTED) {
+            requestPermissionsIfNecessary(new String[]{
+                    Manifest.permission.ACCESS_FINE_LOCATION
+            });
+        }
+
+        final boolean gpsEnabled = locationManager.isProviderEnabled(LocationManager.GPS_PROVIDER);
+        final boolean networkEnabled = locationManager.isProviderEnabled(LocationManager.NETWORK_PROVIDER);
+
+        if(gpsEnabled)
+            locationManager.requestLocationUpdates(LocationManager.GPS_PROVIDER, 0, 0, locationListenerGps);
```

Comment:
Here it would be nice, if the user can control the minimum distance/time via widget parameters to reduce the drain on the device battery.

Reply:
with the fused provider it is slightly different. I will try to export these parameters in the widget if we decide to keep this version using the `LocationManager`.